### ISSUE TITLE
extract paragraph with below table 

### ISSUE THREAD

**Issue by danny007in** [pymupdf/PyMuPDF#2699]
PDF is http://kea.kar.nic.in/diploma_2022/engg_cutoff_gen_mock.pdf

![image](https://github.com/pymupdf/PyMuPDF/assets/50356015/666795d7-70e0-4a22-86fa-767bc18b8a5b)

while extracting all college name is coming and tables are separated, but require line by line to export in excel

**Comment by JorjMcKie** [comment]
Sorry, I don't understand what this means:

> while extracting all college name is coming and tables are separated, but require line by line to export in excel

**Comment by JorjMcKie** [comment]
After a closer look I saw that you **_may_** be hit by a bug that someone detected and submitted a fix for today.

The table finder has a problem telling apart the many tables on your pages, because there are all sorts of line drawings outside tables, which are confusing it.

My recommendation:
1. Apply a hotfix by manually modifying `table.py` in your installation as suggested [here](https://github.com/pymupdf/PyMuPDF/pull/2702). Essentially delete line 1893:
![grafik](https://github.com/pymupdf/PyMuPDF/assets/8290722/13e18415-c5a1-4db7-b40a-2d84f35ad522)

2. Use the following script to detect all tables. The script uses certain table headers as table delimiters. Then feeds these sub rectangles into the table finder.

```python
import fitz

doc = fitz.open("engg_cutoff_gen_mock.pdf")

for page in doc:
    page.clean_contents()
    print(f"Locating tables on {page}.")
    prect = page.rect

    # Locate words "Ennn" and use their bottom y as table delimiter
    y_coords = [
        w[3]  # bottom coordinate of word
        for w in page.get_text("words")
        if w[4].startswith("E") and w[4][1:].isdigit() and len(w[4]) == 4
    ] + [
        prect.y1
    ]  # always append page bottom coordinate

    clips = []  # look for tables inside these clips

    # compute the clips
    for i in range(len(y_coords) - 1):
        clip = +prect
        clip.y0 = y_coords[i]
        clip.y1 = y_coords[i + 1]
        clips.append(clip)

    # now we hopefully are equipped to find all tables!
    for i, clip in enumerate(clips):
        tabs = page.find_tables(clip=clip)
        if tabs.tables:
            tab = tabs[0]
            page.draw_rect(tab.bbox, color=(1, 0, 0))
            print(f"   Detected table {i}")

doc.ez_save("x.pdf")
```

**Comment by danny007in** [comment]
Hi @JorjMcKie 
me new to python, but my suggestion is to implement linter to this project like https://github.com/pylint-dev/pylint

**Comment by danny007in** [comment]
Only first table border is red

![image](https://github.com/pymupdf/PyMuPDF/assets/50356015/a827be09-ec88-4e2c-9e19-44a6dbb4612f)

**Comment by JorjMcKie** [comment]
Did you make the suggested changes in table.py?
If not, then the tables after the first one are not identified.

**Comment by danny007in** [comment]
> Did you make the suggested changes in table.py? If not, then the tables after the first one are not identified.

updated `table.py`, even though it was not working, after refreshing jupyter and vscode (cache issue), its working now

**Comment by danny007in** [comment]
is there any method that i can easily organize table

```
[
  {
    "college_code": "E001",
    "college_name": "E001 University of Visvesvaraya College of Engineering  Bangalore",
    "college_district": "BANGALORE",
    "course": "AI Artificial Intelligence",
    "cast": "1G",
    "rank": 7418
  },
]
```

**Comment by JorjMcKie** [comment]
> is there any method that can easily organize table
> 
> ```
> [
>   {
>     "college_code": "E001",
>     "college_name": "E001 University of Visvesvaraya College of Engineering  Bangalore",
>     "college_district": "BANGALORE",
>     "course": "AI Artificial Intelligence",
>     "cast": "1G",
>     "rank": 7418
>   },
> ]
> ```

Sure!
If `tab` is your table, then `df = tab.to_pandas(https://pypi.org/project/pandas/)` creates a [pandas]() DataFrame that does all that for you!

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.5.
